Plugins/BenchmarkTool/BenchmarkTool.swift

```diff
@@ -113,11 +113,6 @@ struct BenchmarkTool: AsyncParsableCommand {
             print(reason)
             print("")
         }
-        #if canImport(Darwin)
```

Comment:
Can't remove this, as CI workflows for threshold violations would break as the proper exit code is no longer propagated. This is called from call sites with e.g. `exitCode: .thresholdViolation`. 

You need to keep track of the exitCode and return it with an appropriate `exit()` at the end of `run()` of the BenchmarkTool (it's useful for CI to understand both if some benchmark failed, or if it was an automated threshold check violation). In that case I'd say thresholdViolation > error > success as we only can return a single value.

Reply:
That makes sense, I made some changes to keep the threshold throwing, while also allowing benchmarks to continue if there is a failed benchmark. If the user is trying to save the results to a file, I added printing which benchmarks failed.

Reply:
Good point, I added a switch case to handle each error code appropriately and a default